manifests/charts/base/templates/role.yaml

```diff
@@ -18,3 +18,20 @@ rules:
   resources: ["secrets"]
   # TODO lock this down to istio-ca-cert if not using the DNS cert mesh config
   verbs: ["create", "get", "watch", "list", "update", "delete"]
+
+{{- if or .Values.global.externalIstiod .Values.global.centralIstiod }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: istio-reader-{{ .Values.global.istioNamespace }}
+  labels:
+    app: istio-reader
+    release: {{ .Release.Name }}
+rules:
+  # permissions to verify the webhook is ready and rejecting
+  # invalid config. We use --server-dry-run so no config is persisted.
+  - apiGroups: ["networking.istio.io"]
+    verbs: ["create"]
+    resources: ["gateways"]
```

Comment:
ideally we would have a separate non reader role for this, perhaps in a follow up PR?

Reply:
Also, wonder if this is really needed if we terminate the connection at the gw and have real certs at the gw.